ipcs/socket/socket.go

```diff
@@ -83,28 +83,39 @@ func (s *Socket) Listen() error {
 
 // Send writes the given message to all connection clients
 func (s *Socket) Send(msg []byte) error {
-	// Prefix the message with an 8 byte length
-	lenBytes := make([]byte, 8)
-	binary.BigEndian.PutUint64(lenBytes, uint64(len(msg)))
-	msg = append(lenBytes, msg...)
-
+	var conns []net.Conn = nil
 	// Get a copy of connections
 	s.connLock.RLock()
-	conns := make([]net.Conn, len(s.conns))
-	i := 0
-	for conn := range s.conns {
-		conns[i] = conn
-		i++
+	if len(s.conns) > 0 {
+		conns = make([]net.Conn, len(s.conns))
+		i := 0
+		for conn := range s.conns {
+			conns[i] = conn
+			i++
+		}
 	}
 	s.connLock.RUnlock()
 
 	// Write to each connection
+	if len(conns) == 0 {
+		return nil
+	}
+
 	var err error
 	errs := wrappers.Errs{}
+	// Prefix the message with an 8 byte length
+	lenBytes := make([]byte, 8)
```

Comment:
I wonder if this could be allocated on the stack somehow.

Reply:
[8]byte appears to break binary.BigEndian.PutUint64(lenBytes, uint64(len(msg)))

Reply:
If you change lenBytes to a `[8]byte` then you should be able to perform:
```
binary.BigEndian.PutUint64(lenBytes[:], uint64(len(msg)))
```
Similarly below you could range over:
```
[][]byte{lenBytes[:], msg}
```